config/configopaque/opaque.go

```diff
@@ -15,16 +15,19 @@
 package configopaque // import "go.opentelemetry.io/collector/config/configopaque"
 
 import (
-	"bytes"
 	"encoding"
 )
 
 // String alias that is marshaled in an opaque way.
 type String string
 
+const maskedString = "********"
```

Comment:
```suggestion
const maskedString = "[REDACTED]"
```

I think I would prefer to have a placeholder that makes clear no length information is carried.  Users may still assume that `********` masks an eight character string.

Reply:
Done.